src/Type/Definition/HasFieldsTypeImpl.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+namespace GraphQL\Type\Definition;
+
+use GraphQL\Error\InvariantViolation;
+use GraphQL\Utils\Utils;
+
+use function array_keys;
+
+trait HasFieldsTypeImpl
+{
+    /**
+     * Lazily initialized.
+     *
+     * @var array<string, FieldDefinition>
+     */
+    private array $fields;
+
+    /**
+     * @throws InvariantViolation
+     */
+    public function getField(string $name): FieldDefinition
+    {
+        Utils::invariant($this->hasField($name), 'Field "%s" is not defined for type "%s"', $name, $this->name);
+
+        return $this->tryGetField($name);
+    }
+
+    public function tryGetField(string $name): ?FieldDefinition
```

Comment:
what about using `get` for `FieldDefinition` and `find` for `FieldDefinition|null`?

```suggestion
    public function findField(string $name): ?FieldDefinition
```

Reply:
Sounds good to me!